mmv1/third_party/terraform/utils/transport.go.erb

```diff
@@ -50,7 +50,9 @@ func sendRequestWithTimeout(config *Config, method, project, rawurl, userAgent s
 	reqHeaders.Set("User-Agent", userAgent)
 	reqHeaders.Set("Content-Type", "application/json")
 
-	if config.UserProjectOverride && project != "" {
+	if !config.UserProjectOverride {
+		reqHeaders.Set("X-Goog-User-Project", "")
```

Comment:
Why is this necessary?

Reply:
What is the value of explicitly setting the "X-Goog-User-Project" header to empty? It seems like the current behavior should be to not set it at all, which I would expect to be enough - do we also need to make sure that it gets unset if it was previously set somewhere else?

Reply:
Current behavior is that when environment variables `USER_PROJECT_OVERRIDE` is set to true and `GOOGLE_BILLING_PROJECT` is set to some project, "X-Goog-User-Project" header is set to `GOOGLE_BILLING_PROJECT`. https://github.com/hashicorp/terraform-provider-google/blob/26ff814b39d07b56b657eeb92e0abf423077be58/google/config.go#L396-L398

	if c.UserProjectOverride && c.BillingProject != "" {
		headerTransport.Set("X-Goog-User-Project", c.BillingProject)
	}

The issue is that when "X-Goog-User-Project" header is set to `GOOGLE_BILLING_PROJECT`, an error occurs for the resource `google_client_openid_userinfo` specifically and not for other resources. To avoid the error,  I set  UserProjectOverride to false before the API request to get the user info 

	origUserProjectOverride := config.UserProjectOverride
	config.UserProjectOverride = false

and then the header "X-Goog-User-Project" is set to empty with false UserProjectOverride.
	
     if !config.UserProjectOverride {
		reqHeaders.Set("X-Goog-User-Project", "")
    }


To not affect other resources,   reset UserProjectOverride to original value after the API request for the resource `google_client_openid_userinfo`. 

       config.UserProjectOverride = origUserProjectOverride

Reply:
> If config.UserProjectOverride is set to false, the block calling reqHeaders.Set("X-Goog-User-Project", project) will already be skipped, which means that the header won't be sent at all. That would be the same behavior that calls to GetCurrentUserEmail with no project set currently have in the provider, because they already aren't setting the header at all.

Actually when the environment variable `USER_PROJECT_OVERRIDE` is set to true and `GOOGLE_BILLING_PROJECT` is set to some project, first the header `X-Goog-User-Project` is set in provider config. 
https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/utils/config.go.erb#L307-L311


And then for some resources, the header `X-Goog-User-Project` is set again in resource level (this block of code). 

      if config.UserProjectOverride && project != "" {
		  // Pass the project into this fn instead of parsing it from the URL because
		  // both project names and URLs can have colons in them.
		  reqHeaders.Set("X-Goog-User-Project", project)
      }

If config.UserProjectOverride is set to false, the block calling reqHeaders.Set("X-Goog-User-Project", project) will already be skipped. That is right. But  the header `X-Goog-User-Project` is already set in the provider level.
To  override the header `X-Goog-User-Project` to empty in resource level, I added the code block

	if !config.UserProjectOverride {
		reqHeaders.Set("X-Goog-User-Project", "")
       }

Reply:
Thank you for the detailed explanation, and for writing it out twice - I see that you explained that above and I misunderstood it. Sorry about that, and thanks again!